pynest/nest/tests/test_refractory.py

```diff
@@ -0,0 +1,271 @@
+
+import unittest
+
+import numpy as np
+
+import nest
+
+"""
+Assert that all neuronal models that have a refractory period implement it
+correctly (except for Hodgkin-Huxley models which cannot be tested).
+
+Details
+-------
+Submit the neuron to a constant excitatory current so that it spikes in the
+[0, 50] ms.
+A ``spike_detector`` is used to detect the time at which the neuron spikes and
+a ``voltmeter`` is then used to make sure the voltage is clamped to ``V_reset``
+during exactly ``t_ref``.
+
+For neurons that do not clamp the potential, use a very large current to
+trigger immediate spiking
+
+Untested models
+---------------
+* ``aeif_cond_alpha_RK5``
+* ``ginzburg_neuron``
+* ``hh_cond_exp_traub``
+* ``hh_psc_alpha``
+* ``hh_psc_alpha_gap``
+* ``ht_neuron``
+* ``iaf_chs_2007``
+* ``iaf_chxk_2008``
+* ``iaf_tum_2000``
+* ``izhikevich``
+* ``mcculloch_pitts_neuron``
+* ``parrot_neuron``
+* ``parrot_neuron_ps``
+* ``pp_pop_psc_delta``
+* ``pp_psc_delta``
+* ``sli_neuron``
+"""
+
+
+# --------------------------------------------------------------------------- #
+#  Models, specific parameters
+# -------------------------
+#
+
+# list of all neuronal models that can be tested by looking at clamped V
+neurons_V_clamped = [
+    'aeif_cond_alpha',
+    'aeif_cond_alpha_multisynapse',
+    'aeif_cond_beta_multisynapse',
+    'aeif_cond_exp',
+    'aeif_psc_alpha',
+    'aeif_psc_exp',
+    'gif_cond_exp',
+    'gif_cond_exp_multisynapse',
+    'gif_psc_exp',
+    'gif_psc_exp_multisynapse',
+    'iaf_cond_alpha',
+    'iaf_cond_alpha_mc',
+    'iaf_cond_exp',
+    'iaf_cond_exp_sfa_rr',
+    'iaf_neuron',
+    'iaf_psc_alpha',
+    'iaf_psc_alpha_multisynapse',
+    'iaf_psc_delta',
+    'iaf_psc_exp',
+    'iaf_psc_exp_multisynapse',
+]
+
+# neurons that must be tested through a high current to spike immediately
+# (t_ref = interspike)
+neurons_interspike = [
+    "amat2_psc_exp",
+    "mat2_psc_exp",
+    "ht_neuron",
+]
+
+neurons_interspike_ps = [
+    "iaf_psc_alpha_canon",
+    "iaf_psc_alpha_presc",
+    "iaf_psc_delta_canon",
+    "iaf_psc_exp_ps",
+]
+
+# models that cannot be tested
+ignore_model = [
+    "aeif_cond_alpha_RK5",  # this one is faulty and will be removed
+    "ginzburg_neuron",
+    "hh_cond_exp_traub",
+    "hh_psc_alpha",
+    "hh_psc_alpha_gap",
+    "iaf_chs_2007",
+    "iaf_chxk_2008",
+    "iaf_tum_2000",
+    "izhikevich",
+    "mcculloch_pitts_neuron",
+    "parrot_neuron",
+    "parrot_neuron_ps",
+    "pp_pop_psc_delta",
+    "pp_psc_delta",
+    "sli_neuron",
+]
+
+tested_models = [m for m in nest.Models("nodes") if (nest.GetDefaults(
+                 m, "element_type") == "neuron" and m not in ignore_model)]
+
+# additional parameters for the connector
+add_connect_param = {
+    "iaf_cond_alpha_mc": {"receptor_type": 7},
+}
+
+
+# --------------------------------------------------------------------------- #
+#  Simulation and refractory time
+# -------------------------
```

Comment:
you could change this to "refractory time limits" to avoid confusion as this are only the lower and upper bounds right?